x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java

```diff
@@ -129,23 +130,45 @@ public SnapshotRetentionConfiguration getRetentionPolicy() {
         return this.retentionPolicy;
     }
 
-    public long calculateNextExecution() {
-        final Cron scheduleEvaluator = new Cron(this.schedule);
-        return scheduleEvaluator.getNextValidTimeAfter(System.currentTimeMillis());
+    /**
+     * @return whether `schedule` is a cron expression, as opposed to a time unit
+     */
+    public boolean isCronSchedule() {
```

Comment:
Ehhhhhhhhh, running a try/catch here each time somebody invokes this kinda rubs me the wrong way. I would feel a bit better about it if it were executed once in the constructor and remembered forever.